src/DocumentFormat.OpenXml.CodeGeneration.Linq/Program.cs

```diff
@@ -0,0 +1,536 @@
+                    }
+
+                    if (elementAttributeQualifiedNames.Any())
+                    {
+                        string csv = string.Join(", ", elementAttributeQualifiedNames.Select(See));
+                        output.WriteLine($"        /// <item><description>has the following XML attributes: {csv}.</description></item>");
+                    }
+
+                    if (elementClassNames.Any())
+                    {
+                        string csv = string.Join(", ", elementClassNames);
+                        output.WriteLine($"        /// <item><description>corresponds to the following strongly-typed classes: {csv}.</description></item>");
+                    }
+
+                    output.WriteLine(@"        /// </list>");
+                }
+
+                if (attributes != string.Empty)
+                {
+                    output.WriteLine(@"        /// <para>As an XML attribute, it:</para>");
+                    output.WriteLine(@"        /// <list type=""bullet"">");
+
+                    if (attributeContainerQualifiedNames.Any())
+                    {
+                        string csv = string.Join(", ", attributeContainerQualifiedNames.Select(See));
+                        output.WriteLine($"        /// <item><description>is contained in the following XML elements: {csv}.</description></item>");
+                    }
+
+                    if (attributePropertyNames.Any())
+                    {
+                        string csv = string.Join(", ", attributePropertyNames);
+                        output.WriteLine($"        /// <item><description>corresponds to the following strongly-typed properties: {csv}.</description></item>");
+                    }
+
+                    output.WriteLine(@"        /// </list>");
+                }
+
+                if (hasRemarks)
+                {
+                    output.WriteLine(@"        /// </remarks>");
+                }
+
+                // Emit field declaration.
+                output.WriteLine($"        public static readonly XName {info.FieldName} = {fieldRhsPrefix}\"{info.LocalName}\";");
+            }
+
+            // End class declaration.
+            output.WriteLine(@"    }");
+        }
+
+        private static IReadOnlyCollection<ElementMetadata> AssembleElementMetadata(
+            IDictionary<OpenXmlQualifiedName, FieldInfo> fieldInfos)
+        {
+            var visitedElementTypes = new HashSet<Type>();
+            var elementMetadataCollection = new List<ElementMetadata>();
+
+            foreach (ElementLookup.ElementChild elementChild in ElementLookup.Parts.Elements)
+            {
+                AssembleElementMetatata(ElementMetadata.None, elementChild, visitedElementTypes, elementMetadataCollection, fieldInfos);
+            }
+
+            return elementMetadataCollection;
```

Comment:
>  Error CS0266: Cannot implicitly convert type 'System.Collections.Generic.List<DocumentFormat.OpenXml.Framework.Metadata.ElementMetadata>' to 'DocumentFormat.OpenXml.IReadOnlyCollection<DocumentFormat.OpenXml.Framework.Metadata.ElementMetadata>'

This breaks on .net 3.5/.net 4.0

Reply:
It does not have to work on .net 3.5/.net 4.0, does it? This is just a build tool that generates code and can thus target any appropriate .net framework or .net 5.0. You would only run this **once** to generate the namespace-related classes.

Reply:
The way the CI is currently set up is to cross-compile various permutations in order to get some test coverage on the .net 3.5/.net 4.0 builds. It needs to build for those targets at the moment, but you can do a compile remove or something so that those builds are no-op.

With VS 2022 supporting larger projects, I'd like to see a simplification of the build process....

Reply:
I've changed the project to target the `$(SamplesFrameworks)`, i.e., `net46`, `netcoreapp2.1`, `netcoreapp3.1`, and `net5.0`. I would hope that this is enough.